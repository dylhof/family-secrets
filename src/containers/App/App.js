import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Header from '../Header/Header';
import NavBar from '../NavBar/NavBar';
import Error from '../../components/Error/Error';
import Loading from '../../components/Loading/Loading';
import CardArea from '../../containers/CardArea/CardArea';


export class App extends Component {

  render() {
    const { error, isLoading} = this.props
    return (
      <div className="App">
        <Header/>
        <NavBar/>
        <CardArea/>
        {error && <Error/>}
        {isLoading && <Loading/>}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  error: state.error,
  isLoading: state.isLoading
});

export default connect(mapStateToProps)(App);

App.propTypes = {
  error: PropTypes.string,
  isLoading: PropTypes.bool
}