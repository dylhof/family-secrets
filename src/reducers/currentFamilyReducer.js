export const currentFamilyReducer = (state = '', action) => {
  switch(action.type) {
    case 'SET_CURRENT_FAMILY':
      return action.currentFamily
    default: 
      return state
  };
};