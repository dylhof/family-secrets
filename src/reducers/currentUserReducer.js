export const currentUserReducer = (state=null, action) => {
  switch(action.type) {
    case 'SET_CURRENT_USER':
      const currentUser = {id: action.id, name: action.name, email: action.email}
      return currentUser
    case 'LOG_OUT':
      return null
    default:
      return state
  };
};