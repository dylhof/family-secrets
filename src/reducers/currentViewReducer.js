export const currentViewReducer = (state = '', action) => {
  switch (action.type) {
    case 'SET_CURRENT_VIEW':
      return action.currentView
    default:
      return state;
  };
};