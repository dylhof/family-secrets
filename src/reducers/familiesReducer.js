export const familiesReducer = (state = [], action) => {
  switch(action.type) {
    case 'SET_FAMILIES':
      return action.families;
    default: 
      return state;
  }
}