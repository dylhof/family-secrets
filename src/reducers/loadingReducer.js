export const loadingReducer = (state = false, action) => {
  switch (action.type) {
    case 'SET_IS_LOADING':
      return action.isLoading
    default:
      return state
  };
};