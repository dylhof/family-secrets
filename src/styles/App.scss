body {
  background-color: $body-color;
}

.App {
  display: grid;
  grid-template-rows: 100px 50px 1fr;
  grid-template-areas: 
    'header',
    'navbar',
    'body';
}