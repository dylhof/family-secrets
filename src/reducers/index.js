import { combineReducers } from 'redux';
import { errorReducer } from './errorReducer';
import { loadingReducer } from './loadingReducer';
import { currentViewReducer } from './currentViewReducer';
import { currentFamilyReducer } from './currentFamilyReducer';
import { currentUserReducer } from './currentUserReducer';


export const rootReducer = combineReducers({
  error: errorReducer,
  isLoading: loadingReducer,
  currentView: currentViewReducer,
  currentFamily: currentFamilyReducer,
  currentUser: currentUserReducer
});