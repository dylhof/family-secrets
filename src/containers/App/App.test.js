import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import {App, mapStateToProps} from './App';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { rootReducer } from '../../reducers';


describe('App', () => {
  const mockProps = {
    error: '',
    isLoading: false
  }

  it('renders without crashing', () => {
    const div = document.createElement('div');
    const store = createStore(rootReducer)
    ReactDOM.render(<Provider store={store}><App /></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('App Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App {...mockProps}/>)
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });

    it('should match the snapshot with an error in props', () => {
      wrapper = shallow(<App {...mockProps} error='Oh no!' />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when isLoading is true', () => {
      wrapper = shallow(<App {...mockProps} isLoading={true}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
   
    it('should return an object with an error and isLoading', () => {
      const mockState = {
        error: 'Oh no! Error!',
        isLoading: false,
        otherState: 'something'
      };
      const expected = {
        error: 'Oh no! Error!',
        isLoading: false,
      };
      const result = mapStateToProps(mockState);
      expect(result).toEqual(expected);
    });
  });
});
