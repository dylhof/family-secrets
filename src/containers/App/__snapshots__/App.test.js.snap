// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot 1`] = `
<div
  className="App"
>
  <Connect(Header) />
  <Connect(NavBar) />
  <Connect(CardArea) />
</div>
`;

exports[`App App Component should match the snapshot when isLoading is true 1`] = `
<div
  className="App"
>
  <Connect(Header) />
  <Connect(NavBar) />
  <Connect(CardArea) />
  <Loading />
</div>
`;

exports[`App App Component should match the snapshot with an error in props 1`] = `
<div
  className="App"
>
  <Connect(Header) />
  <Connect(NavBar) />
  <Connect(CardArea) />
  <Error />
</div>
`;
